apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -234,13 +243,41 @@ if (true) {
   );
 }
 const calcomAdapter = CalComAdapter(prisma);
+const getDerivedEncryptionKey = async (secret: BinaryLike | KeyObject) => {
+  return await hkdfSync("sha256", secret, "", "NextAuth.js Generated Encryption Key", 32);
+};
 export default NextAuth({
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore
   adapter: calcomAdapter,
   session: {
     strategy: "jwt",
   },
+  jwt: {
+    encode: async ({ secret, token }) => {
+      const encryptionSecret = await getDerivedEncryptionKey(secret);
+      return await new jose.EncryptJWT({
+        sub: token?.sub,
+        name: token?.name,
+        email: token?.email,
+      })
+        .setProtectedHeader({
+          alg: "dir",
+          enc: "A256GCM",
+        })
+        .setIssuedAt()
+        .setExpirationTime(token && token.sessionTimeout ? `${token.sessionTimeout}m` : "30d")
+        .setJti(uuidv4())
+        .encrypt(new Uint8Array(encryptionSecret));
+    },
+    decode: async ({ secret, token }) => {
+      const encryptionSecret = await getDerivedEncryptionKey(secret);
+      const { payload } = await jose.jwtDecrypt(token || "", new Uint8Array(encryptionSecret), {
+        clockTolerance: 15,
+      });
+      return payload;
+    },
+  },
```

Comment:
This way of encrypting and decrypting the token is the same as `next-auth` does it internally by default when no encode/decode functions are implemented.

Reply:
@leog Shouldn't maxAge option for next-auth do the trick here?

Reply:
I was think the same thing @alannnc aren't we kinda over engineering this?

Reply:
@alannnc you mean the session.maxAge property in NextAuth config object? Couldn’t find a more straightforward way to configure it grabbing the value from the user metadata. Perhaps I can simplify the logic making a DB query from jwt.encode function given the token.sub as userId or token.email and just setting the expiration from it instead of going the user property approach.

Reply:
That's what I mean
> Perhaps I can simplify the logic making a DB query from jwt.encode function given the token.sub as userId or token.email and just setting the expiration from it instead of going the user property approach.

This sounds like a more easy way to do it.

Reply:
It would also mean an additional db query each time a token needs to be encoded, which is why I went for the user property approach. It would simplify but could take a toll in performance

Reply:
We don't mind the extra query as it remains simple and robust. Also we're going to adopt DB sessions and accounts soon enough. So we're going to query anyways.

Reply:
Gotcha. Brought back my first implementation then.